### DISCUSSION TITLE
UICollector from Tasker

### DISCUSSION THREAD

**Discussion by AJDempsey** [NightscoutFoundation/xDrip#2908] [Ideas]
Hi I'm using the Dexcom ONE and my care team use clarity so I can't totally switch to xDrip+ for my care.
Currently the Dexcom ONE notification just displays "Dexcom app is running" rather than anything actually useful. Would it be possible to extend https://github.com/NightscoutFoundation/xDrip/blob/master/app/src/main/java/com/eveningoutpost/dexdrip/services/UiBasedCollector.java#L64 to include tasker/autonotifications

**Comment by jamorham** [comment]
I'm not sure from your message what the use case is here, but have you tried using the `Companion Bluetooth` preference option? Does that not work already?

**Reply by AJDempsey** [reply]
No the Companion Bluetooth setting doesn't work for me unfortunately

![Screenshot_20230627-200432](https://github.com/NightscoutFoundation/xDrip/assets/1737807/c3078479-c2d7-4301-a75d-6eab35127166)

The screenshot hopefully shows what I mean. My dexcom One app doesn't show a BG reading in the notification so I've been generating my own notification using tasker. It's not great but it's helpful for a quick glance, the xDrip+ alarms are much better than Dexcom ONEs alarms and I want to keep sharing through dexcom clarity so I can't move to using xDrip+ as the app connected to the sensor.

**Reply by jamorham** [reply]
Please try the latest nightly and let me know if it now works.

**Reply by AJDempsey** [reply]
Perfect that's it work now, thank you so much!

**Reply by Navid200** [reply]
@AJDempsey 
If you don't mind telling me more detail about how you do this, I will add it to documentation so that others can use your method as well.
If you want, I have no problem adding your name, or user ID, at the beginning of the instructions for credit.

**Reply by AJDempsey** [reply]
@Navid200
The UIBasedcollector started working without the need for tasker. The actual tasker files are fairly basic, it just launches the app, scrapes the screen using autoinput, and then updates a notification with the info using autonotification. There's one for when the phone is being used and one for when the screen is off

`    Profile: Dexcom Notification Bodge
    	Time:  Every 5m
    
    
    
    Enter Task: Dexcom Scrape
    
    A1: Test Display [
         Type: Is Locked
         Store Result In: %locked ]
    
    A2: If [ %locked neq true ]
    
        A3: App Info [ ]
    
        A4: Turn On [
             Block Time (Check Help): 999 ]
    
        A5: AutoNotification Actions [
             Configuration: Notification Apps: Dexcom ONE
             Notification Title: Dexcom app is running (case ins)
             Timeout (Seconds): 20
             Structure Output (JSON, etc): On ]
    
        A6: AutoInput UI Query [
             Configuration: App Package: com.dexcom.dexcomone
             Variables: mmolval
             Timeout (Seconds): 5
             Structure Output (JSON, etc): On
             Continue Task After Error:On ]
    
        A7: If [ %app_name Set ]
    
            A8: Load Last App
    
        A9: Else
    
            A10: Go Home [
                  Page: 0 ]
    
        A11: End If
    
        A12: Vibrate Mode [
              Mode: Off ]
    
        A13: Sound Mode [
              Mode: Mute ]
    
        A14: AutoNotification [
              Configuration: Title: %mmolval mmol/L - %TIME
             Status Bar Text Size: 16
             Id: %aiapp
             Priority: 2
             Visibility: Public
             Separator: ,
              Timeout (Seconds): 1
              Structure Output (JSON, etc): On ]
    
        A15: Sound Mode [
              Mode: Vibrate ]
    
        A16: Vibrate Mode [
              Mode: Vibrate ]
    
    A17: End If
    
    `
`    Profile: Dexcom Screen Off
    	Event: Display On
    
    
    
    Enter Task: Dexcom Notification Wake
    
    A1: AutoNotification Actions [
         Configuration: Notification Apps: Dexcom ONE
         Notification Title: Dexcom app is running (case ins)
         Timeout (Seconds): 20
         Structure Output (JSON, etc): On ]
    
    A2: AutoInput UI Query [
         Configuration: App Package: com.dexcom.dexcomone
         Variables: mmolval
         Timeout (Seconds): 5
         Structure Output (JSON, etc): On
         Continue Task After Error:On ]
    
    A3: Vibrate Mode [
         Mode: Off ]
    
    A4: Sound Mode [
         Mode: Mute ]
    
    A5: AutoNotification [
         Configuration: Title: %mmolval mmol/L - %TIME
         Status Bar Text Size: 16
         Id: %aiid()
         Priority: 2
         Visibility: Public
         Separator: ,
         Update Notification: true
         Timeout (Seconds): 20
         Structure Output (JSON, etc): On
         Continue Task After Error:On ]
    
    A6: Sound Mode [
         Mode: Vibrate ]
    
    A7: Vibrate Mode [
         Mode: Vibrate ]
    
    A8: Go Home [
         Page: 0 ]
    
    `

**Reply by Navid200** [reply]
@AJDempsey Thanks for all the detail.

But, this is what makes your response very confusing for me.

> The UIBasedcollector started working without the need for tasker. 

So, what does that mean?
xDrip is showing readings now with no need to the tasker, even though Dexcom app is still not showing readings in its notifications?!

**Reply by AJDempsey** [reply]
![Screenshot_20230702-000155](https://github.com/NightscoutFoundation/xDrip/assets/1737807/57e6522f-b1f0-4813-93b7-8d6e991eb418)

Yeah, not sure exactly how it's working but with the Bluetooth companion setting on and using the UI collector it's working fine now

**Reply by Navid200** [reply]
Fantastic
